### PR TITLE
Make sure to escape error messages

### PR DESCRIPTION
**Pull request by tsteur**
Just in case an exception that is displayed there contains any user input

### CODE REVIEW

File: core/testMinimumPhpVersion.php [hunk 1]
```diff
@@ -111,6 +111,22 @@ function Piwik_GetErrorMessagePage($message, $optionalTrace = false, $optionalLi
...
+        $message = str_replace(array("<br />", "<br>", "<br/>", "</p>"), "\n", $message);
+        $message = str_replace("\t", "", $message);
+        $message = strip_tags($message);
+
+        if ($isCli === null) {
+            $isCli = PHP_SAPI == 'cli';
+        }
+
+        if (!$isCli) {
+            if (version_compare('5.2.3', PHP_VERSION) > 0) {
```
Comment (mattab) [inline core/testMinimumPhpVersion.php]: version_compare should be `< 0`

Comment (tsteur) [inline core/testMinimumPhpVersion.php]: fixed

**Comment by mattab** [comment]
Looks good to me

**Comment by mattab** [comment]
Actually there is a regression where we lose the bold message in: http://builds-artifacts.piwik.org/ui-tests.master/13711.7/UIIntegrationTest_db_connect_error

 * we need to strip_tags and remove `<br>` only when in CLI

**Comment by tsteur** [comment]
> we need to strip_tags and remove <br> only when in CLI

Then htmlentites would escape them and the html would be visible. Looking deeper at the code this method `Piwik_GetErrorMessagePage` is only used in 2 or 3 parts and the one that contains user input is kinda escaped before already (https://github.com/piwik/piwik/blob/2.14.0-b10/core/ExceptionHandler.php#L70-L72) and doc block says `@param string $message Main message, must be html encoded before calling`

**Pull request merged**